ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,512 @@
+    if expected_type in [typing.Any, inspect._empty]:  # type: ignore
+        return
+    if not isinstance(expected_type, typing._GenericAlias):  # type: ignore
+        raise_value_error_if_parameter_has_unexpected_type(
+            parameter=parameter,
+            parameter_name=parameter_name,
+            expected_type=expected_type,
+        )
+        return
+    expected_type_dict = expected_type.__dict__
+    origin_class = expected_type_dict.get("__origin__")
+    nested_elements_class = expected_type_dict.get("__args__")
+    # Union type with nested elements check
+    if origin_class == typing.Union:
+        expected_args = expected_type_dict.get("__args__")
+        checks_counter = 0
+        errors_counter = 0
+        for expected_arg in expected_args:
+            try:
+                checks_counter += 1
+                check_parameter_type(parameter, parameter_name, expected_arg)
+            except ValueError:
+                errors_counter += 1
+        if errors_counter == checks_counter:
+            actual_type = type(parameter)
+            raise ValueError(
+                f"Unexpected type of '{parameter_name}' parameter, expected: {expected_args}, "
+                f"actual type: {actual_type}, actual value: {parameter}"
+            )
+    # Checking parameters with nested elements
+    elif issubclass(origin_class, typing.Iterable):
+        check_nested_classes_parameters(
+            parameter=parameter,
+            parameter_name=parameter_name,
+            origin_class=origin_class,
+            nested_elements_class=nested_elements_class,
+        )
+
+
+def check_input_parameters_type(custom_checks: typing.Optional[dict] = None):
+    """
+    Decorator to check input parameters type
+    :param custom_checks: dictionary where key - name of parameter and value - custom check class
+    """
+    if custom_checks is None:
+        custom_checks = {}
+
+    def _check_input_parameters_type(function):
+        @wraps(function)
+        def validate(*args, **kwargs):
+            # Forming expected types dictionary
+            signature = inspect.signature(function)
+            expected_types_map = signature.parameters
+            if len(expected_types_map) < len(args):
+                raise TypeError("Too many positional arguments")
+            # Forming input parameters dictionary
+            input_parameters_values_map = dict(zip(signature.parameters.keys(), args))
+            for key, value in kwargs.items():
+                if key in input_parameters_values_map:
+                    raise TypeError(
+                        f"Duplication of the parameter {key} -- both in args and kwargs"
+                    )
+                input_parameters_values_map[key] = value
+            # Checking input parameters type
+            for parameter_name in expected_types_map:
+                parameter = input_parameters_values_map.get(parameter_name)
+                if parameter is None:
```

Comment:
Please, fix it
```suggestion
                if parameter_name not in input_parameters_values_map:
```
The reason: we cannot forbid to pass `None` value explicitly as a value of a parameter, e.g `a = dict(x=None, y=None)`

Reply:
thanks, I will fix it in https://github.com/openvinotoolkit/training_extensions/pull/891